.ci/azure-pipelines/build_macos.bash

```diff
@@ -1,25 +1,50 @@
 #!/bin/bash -ex
 #
-# Copyright 2019-2020 The Mumble Developers. All rights reserved.
+# Copyright 2019-2021 The Mumble Developers. All rights reserved.
 # Use of this source code is governed by a BSD-style license
 # that can be found in the LICENSE file at the root of the
 # Mumble source tree or at <https://www.mumble.info/LICENSE>.
 #
+# Builds Mumble using the specified build script.
+# The path to the script is relative to the build environment's root.
+# The configuration we build with is adjusted to be close to
+# our release builds.
+#
 # Below is a list of configuration variables used from environment.
 #
-#  BUILD_BINARIESDIRECTORY - The local path on the agent that can be used
-#                            as an output folder for compiled binaries.
-#  BUILD_SOURCESDIRECTORY  - The local path on the agent where the
-#                            repository is downloaded.
+# Predefined variables:
+#
+#  BUILD_BINARIESDIRECTORY      - The local path on the agent that can be used
+#                                 as an output folder for compiled binaries.
+#  BUILD_SOURCESDIRECTORY       - The local path on the agent where the
+#                                 repository is downloaded.
+#
+# Defined in .azure-pipelines.yml:
+#
+#  MUMBLE_ENVIRONMENT_PATH      - Path to the environment's root directory.
+#  MUMBLE_ENVIRONMENT_TOOLCHAIN - Path to the vcpkg CMake toolchain, used for CMake.
+#  MUMBLE_ENVIRONMENT_TRIPLET   - vcpkg triplet.
 #
+# Defined on Azure Pipelines:
+#
+#  BUILD_NUMBER_TOKEN           - Access token for the build number page on our server.
+#
+
+if [[ -n "$BUILD_NUMBER_TOKEN" ]]; then
+	VERSION=$(python "scripts/mumble-version.py" --project)
+	BUILD_NUMBER=$(curl "https://mumble.info/get-build-number?version=$VERSION&token=$BUILD_NUMBER_TOKEN")
+else
+	BUILD_NUMBER=0
+fi
 
-VER=$(python scripts/mumble-version.py)
+VERSION_STRING=$(python "scripts/mumble-version.py")
 
 cd $BUILD_BINARIESDIRECTORY
 
 cmake -G Ninja -DCMAKE_TOOLCHAIN_FILE=$MUMBLE_ENVIRONMENT_TOOLCHAIN -DIce_HOME="$MUMBLE_ENVIRONMENT_PATH/installed/x64-osx" \
-	-DCMAKE_BUILD_TYPE=Release -Dtests=ON -Dversion=$VER -Dstatic=ON -Dsymbols=ON -Donline-tests=ON -Dgrpc=ON \
-	-Ddisplay-install-paths=ON $BUILD_SOURCESDIRECTORY
+      -DCMAKE_BUILD_TYPE=Release -DVERSION_STRING=$VERSION_STRING -DBUILD_NUMBER=$BUILD_NUMBER \
```

Comment:
```suggestion
      -DCMAKE_BUILD_TYPE=Release -DVERSION_STRING=$VERSION_STRING -Dbuild-number=$BUILD_NUMBER \
```

Reply:
Since it is global, shouldn't it be capitalized, anyway? If so, I can adjust my PR after this is merged.

Reply:
More specifically, it's a string instead of an `option`.

Reply:
Well it doesn't really matter if it is an option or not (imo) since options are also global variables (in a sense).
We use lowercase spelling for all other options, so why would we want to change that for the `build-number` variable?

Reply:
I have just seen most practices with CMake where values and options that are modified from the CLI are typically all caps.

Reply:
But we don't follow that already. If we want to do so, I'm fine with that, but that should be done for all variables and options then and in a dedicated commit :point_up:

Reply:
Yeah.